RHDEVDOCS-3304 Document default pipelines SA (#40749)
openshift/openshift-docs: modules/op-disabling-automatic-creation-of-rbac-resources.adoc

// Module included in the following assemblies:
//
// */openshift_pipelines/installing-pipelines.adoc
[id="op-disabling-automatic-creation-of-rbac-resources_{context}"]
= Disabling the automatic creation of RBAC resources

> After default installation, the {pipelines-title} Operator creates multiple Role Based Access Control (RBAC) resources for all namespaces in the cluster. Among them, the `pipelines-scc-rolebinding` SCC role binding resource is a potential security issue, because the associated `pipelines-scc` Security Context Constraint has `RunAsAny` privilege. 

Reviewer: ```suggestion
After default installation, the {pipelines-title} Operator creates multiple Role Based Access Control (RBAC) resources for all namespaces(except openshift-* namespaces) in the cluster. Among them, the `pipelines-scc-rolebinding` SCC role binding resource is a potential security issue, because the associated `pipelines-scc` Security Context Constraint has `RunAsAny` privilege. 
```

Reviewer: For ref: https://github.com/tektoncd/operator/blob/main/pkg/reconciler/common/common.go#L41

> The default installation of the {pipelines-title} Operator creates multiple Role Based Access Control (RBAC) resources for all namespaces in the cluster, except the namespaces matching the `^(openshift|kube)-*` regular expression pattern. Among these RBAC resources, the `pipelines-scc-rolebinding` SCC role binding resource is a potential security issue, because the associated `pipelines-scc` Security Context Constraint has `RunAsAny` privilege. 

Author: It's accurate. See line number 164 and 716 of the source code file https://github.com/tektoncd/operator/blob/main/pkg/reconciler/openshift/tektonconfig/rbac.go

> The default installation of the {pipelines-title} Operator creates multiple Role Based Access Control (RBAC) resources for all namespaces in the cluster, except the namespaces matching the `^(openshift|kube)-*` regular expression pattern. Among these RBAC resources, the `pipelines-scc-rolebinding` SCC role binding resource is a potential security issue, because the associated `pipelines-scc` Security Context Constraint has `RunAsAny` privilege. 

Reviewer 2: @sounix000 I checked. Did not see the * in both the instances in the source code file. Not sure if this is ok. Suggest rechecking once.

> The default installation of the {pipelines-title} Operator creates multiple Role Based Access Control (RBAC) resources for all namespaces in the cluster, except the namespaces matching the `^(openshift|kube)-*` regular expression pattern. Among these RBAC resources, the `pipelines-scc-rolebinding` SCC role binding resource is a potential security issue, because the associated `pipelines-scc` Security Context Constraint has `RunAsAny` privilege. 

Author: Marking as resolve based on our Slack conversation.

> To disable the automatic creation of cluster-wide RBAC resources after the {pipelines-title} Operator is installed, cluster administrators can set the `createRbacResource` parameter to `false` in the cluster level `TektonConfig` custom resource.

Reviewer 2: ```suggestion
To disable the automatic creation of cluster-wide RBAC resources after the {pipelines-title} Operator is installed, cluster administrators can set the `createRbacResource` parameter to `false` in the cluster-level `TektonConfig` custom resource (CR).
```

> .Example `TektonConfig` custom resource

Reviewer 2: ```suggestion
.Example `TektonConfig` CR
```

[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  params:
  - name: createRbacResource
    value: "false"
  profile: all
  targetNamespace: openshift-pipelines
  addon:
    params:
    - name: clusterTasks
      value: "true"
    - name: pipelineTemplates
      value: "true"
...
----

[WARNING]
====

> When you disable automatic creation of the RBAC resources on all namespaces, the default `ClusterTask` resource does not work. For the `ClusterTask` resource to function, cluster administrators or users with appropriate privileges must create the RBAC resources manually for each intended namespace.

Reviewer 2: ```suggestion
As a cluster administrator or user with appropriate privileges, when you disable the automatic creation of the RBAC resources on all namespaces, the default `ClusterTask` resource does not work. For the `ClusterTask` resource to function, you must create the RBAC resources manually for each intended namespace.
```
